components/script/dom/htmllinkelement.rs

```diff
@@ -273,6 +276,13 @@ impl PreInvoke for StylesheetContext {}
 impl AsyncResponseListener for StylesheetContext {
     fn headers_available(&mut self, metadata: Result<Metadata, NetworkError>) {
         self.metadata = metadata.ok();
+        let content_type = self.metadata.clone().and_then(|meta| meta.content_type);
```

Comment:
you are right, it seems like I can use the `if let` and `ref` to get the `content_type`.